jskit/encoders/bi.py

```diff
@@ -101,16 +105,40 @@ def cosine_sim(vec_a: List[float], vec_b: List[float], meta):
 
 
 @jaseci_action(act_group=['bi_enc'], allow_remote=True)
-def infer(contexts: List, candidates: List):
+def infer(contexts: Union[List[str], List[List[float]]], candidates: Union[List[str], List[List[float]]]):
```

Comment:
Let's make it an explicit function parameter what is being passed in. It will make this API easier to pick up and you can use that parameter to do the logic accordingly so you don't have to do all the list data checking stuff. You can do something like this
```
def infer(contexts, candidates, context_type, candidate_type):
```
context_type can be "text" or "embedding"

Reply:
fixed